From 06160bc8e6359948a0e2374fe5486763d28d4632 Mon Sep 17 00:00:00 2001
From: "yinbo.zhu" <yinbo.zhu@nxp.com>
Date: Tue, 15 Aug 2017 16:10:03 +0800
Subject: [PATCH 729/741] USB: fix problems with duplicate filename
 '/bus/platform/devices/fsl-ehci.0'

commit 056479deac1e645d8a5a029908b5af8a678d93b5 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

call trace:
------------[ cut here ]------------
WARNING: CPU: 1 PID: 1 at fs/sysfs/dir.c:31 sysfs_warn_dup+0x68/0x78
sysfs: cannot create duplicate filename '/bus/platform/devices/fsl-ehci.0'
Modules linked in:
CPU: 1 PID: 1 Comm: swapper/0 Not tainted 4.11.0-rc6-next-20170411-00342-g415fab02 #30
Hardware name: Freescale LS1021A
(unwind_backtrace) from [<c040b91c>] (show_stack+0x10/0x14)
[<c040b91c>] (show_stack) from [<c070aa84>] (dump_stack+0x98/0xac)
[<c070aa84>] (dump_stack) from [<c04440bc>] (__warn+0xec/0x104)
[<c04440bc>] (__warn) from [<c044410c>] (warn_slowpath_fmt+0x38/0x48)
[<c044410c>] (warn_slowpath_fmt) from [<c05c0d10>] (sysfs_warn_dup+0x68/0x78)
[<c05c0d10>] (sysfs_warn_dup) from [<c05c0fc0>] (sysfs_do_create_link_sd+0xb4/0xc4)
[<c05c0fc0>] (sysfs_do_create_link_sd) from [<c096aa1c>] (bus_add_device+0xf4/0x18c)
[<c096aa1c>] (bus_add_device) from [<c0968eb4>] (device_add+0x38c/0x574)
[<c0968eb4>] (device_add) from [<c096d09c>] (platform_device_add+0xb8/0x220)
[<c096d09c>] (platform_device_add) from [<c0c503c4>] (fsl_usb2_mph_dr_of_probe+0x420/0x61c)
[<c0c503c4>] (fsl_usb2_mph_dr_of_probe) from [<c096d384>] (platform_drv_probe+0x4c/0xb0)
[<c096d384>] (platform_drv_probe) from [<c096b7c0>] (driver_probe_device+0x238/0x2d8)
[<c096b7c0>] (driver_probe_device) from [<c0969bfc>] (bus_for_each_drv+0x60/0x94)
[<c0969bfc>] (bus_for_each_drv) from [<c096b4a8>] (__device_attach+0xb0/0x114)
[<c096b4a8>] (__device_attach) from [<c096ab38>] (bus_probe_device+0x84/0x8c)
[<c096ab38>] (bus_probe_device) from [<c0968f14>] (device_add+0x3ec/0x574)
[<c0968f14>] (device_add) from [<c096d09c>] (platform_device_add+0xb8/0x220)
[<c096d09c>] (platform_device_add) from [<c0c503c4>] (fsl_usb2_mph_dr_of_probe+0x420/0x61c)
[<c0c503c4>] (fsl_usb2_mph_dr_of_probe) from [<c096d384>] (platform_drv_probe+0x4c/0xb0)
[<c096d384>] (platform_drv_probe) from [<c096b7c0>] (driver_probe_device+0x238/0x2d8)
[<c096b7c0>] (driver_probe_device) from [<c096b90c>] (__driver_attach+0xac/0xb0)
[<c096b90c>] (__driver_attach) from [<c0969b54>] (bus_for_each_dev+0x68/0x9c)
[<c0969b54>] (bus_for_each_dev) from [<c096ade8>] (bus_add_driver+0x1a4/0x21c)
[<c096ade8>] (bus_add_driver) from [<c096c3e4>] (driver_register+0x78/0xf8)
[<c096c3e4>] (driver_register) from [<c0401e30>] (do_one_initcall+0x40/0x168)
[<c0401e30>] (do_one_initcall) from [<c1600df8>] (kernel_init_freeable+0x164/0x200)
[<c1600df8>] (kernel_init_freeable) from [<c0eace38>] (kernel_init+0x8/0x110)
[<c0eace38>] (kernel_init) from [<c0407ca8>] (ret_from_fork+0x14/0x2c)
---[ end trace 7198de059d5332e9 ]---
fsl-usb2-mph-dr fsl-ehci.0: Can't register usb device
fsl-usb2-mph-dr: probe of fsl-ehci.0 failed with error -17
platform fsl-ehci.1: failed to claim resource 0: [mem 0x08600000-0x08600fff]
OF: ERROR: Bad of_node_put() on /soc/usb@8600000
CPU: 1 PID: 1 Comm: swapper/0 Tainted: G        W       4.11.0-rc6-next-20170411-00342-g415fab02 #30
Hardware name: Freescale LS1021A
[<c041084c>] (unwind_backtrace) from [<c040b91c>] (show_stack+0x10/0x14)
[<c040b91c>] (show_stack) from [<c070aa84>] (dump_stack+0x98/0xac)
[<c070aa84>] (dump_stack) from [<c070c074>] (kobject_put+0x74/0xd0)
[<c070c074>] (kobject_put) from [<c096ceac>] (platform_device_release+0x10/0x3c)
[<c096ceac>] (platform_device_release) from [<c0966b88>] (device_release+0x2c/0x90)
[<c0966b88>] (device_release) from [<c070c074>] (kobject_put+0x74/0xd0)
[<c070c074>] (kobject_put) from [<c0c50340>] (fsl_usb2_mph_dr_of_probe+0x39c/0x61c)
[<c0c50340>] (fsl_usb2_mph_dr_of_probe) from [<c096d384>] (platform_drv_probe+0x4c/0xb0)
[<c096d384>] (platform_drv_probe) from [<c096b7c0>] (driver_probe_device+0x238/0x2d8)
[<c096b7c0>] (driver_probe_device) from [<c096b90c>] (__driver_attach+0xac/0xb0)
[<c096b90c>] (__driver_attach) from [<c0969b54>] (bus_for_each_dev+0x68/0x9c)
[<c0969b54>] (bus_for_each_dev) from [<c096ade8>] (bus_add_driver+0x1a4/0x21c)
[<c096ade8>] (bus_add_driver) from [<c096c3e4>] (driver_register+0x78/0xf8)
[<c096c3e4>] (driver_register) from [<c0401e30>] (do_one_initcall+0x40/0x168)
[<c0401e30>] (do_one_initcall) from [<c1600df8>] (kernel_init_freeable+0x164/0x200)
[<c1600df8>] (kernel_init_freeable) from [<c0eace38>] (kernel_init+0x8/0x110)
[<c0eace38>] (kernel_init) from [<c0407ca8>] (ret_from_fork+0x14/0x2c)
fsl-usb2-mph-dr fsl-ehci.0: Can't register usb device
fsl-usb2-mph-dr: probe of fsl-ehci.0 failed with error -16
usbcore: registered new interface driver usb-storage

Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/host/fsl-mph-dr-of.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index 2f07115160f2..da2fab89daad 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -99,9 +99,6 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
-	if (pdata->operating_mode != FSL_USB2_DR_DEVICE)
-		pdev->dev.of_node = ofdev->dev.of_node;
-
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
 		goto error;
-- 
2.17.1

